### ISSUE TITLE
Configuration version and kind are not validated 

### ISSUE THREAD

**Issue by kke** [k0sproject/k0s#1237]
The k0s.yaml "header" fields `apiVersion` and `kind` are not checked.

```shell
$ cat k0s.yaml
apiVersion: hessu
kind: hopo
spec:
  api:
    address: 10.0.0.1
$ k0s validate config -c k0s.yaml
# no error
$ k0s controller -c k0s.yaml
# no error
```

**Comment by mviitane** [comment]
Also these kind of spelling mistakes go through the validation without any error:
```
spec:
  api:
    addressWRONG: 172.1.1.1
    sansWRONG:
    - 172.x.x.x
  storageWRONG:
    type: etcd
    etcd:
      peerAddress: 172.x.x.x
  network:
    podCIDR__WRONG: 10.x.0.0/16

```

**Comment by kke** [comment]
That's odd, because the config is read using the strict yaml parser with `DisallowUnknownFields` enabled 🤔

**Comment by kke** [comment]
The field validation, btw, if it would work, would say json instead of yaml, because the yaml parser from google converts the yaml to json and the decodes it again (for reasons): `error unmarshaling JSON: while decoding JSON: json: unknown field "stringB"`'

**Comment by kke** [comment]
I don't understand why the field validation does not work.

All config loading is done via `ConfigFromString`:

```go
func ConfigFromString(yml string, dataDir string) (*ClusterConfig, error) {
	config := DefaultClusterConfig(dataDir)
	err := strictyaml.YamlUnmarshalStrictIgnoringFields([]byte(yml), config, "interval")
	if err != nil {
		return config, err
	}
	if config.Spec == nil {
		config.Spec = DefaultClusterSpec(dataDir)
	}
	return config, nil
}
```

The strictyaml thing has tests and it does catch unknown fields.

A test like this however fails:

```go
func TestUnknownFieldValidation(t *testing.T) {
	_, err := ConfigFromString(`
apiVersion: k0s.k0sproject.io/v1beta1
kind: ClusterConfig
unknown: 1`, dataDir)

	assert.Error(t, err)
}
```

```
    clusterconfig_types_test.go:41:
        	Error Trace:	clusterconfig_types_test.go:41
        	Error:      	An error is expected but got nil.
        	Test:       	TestUnknownFieldValidation
```

**Comment by kke** [comment]
Aha! Possibly because of the `UnmarshalJSON` function in `ClusterConfig` which uses `json.Unmarshal`.

**Comment by kke** [comment]
Fixed by #1246
